### ISSUE TITLE
negative array access when displaying a ride's excitement rating

### ISSUE THREAD

**Issue by camthesaxman** [OpenRCT2/OpenRCT2#5391]
**OS:** Linux Mint 18.1 (x86-64)
**Version:** 0.0.7
**Commit/Build:** 8f59b04

This does not usually cause a crash, but it has the potential to. When testing a newly created ride with the measurements and test data tab open in the ride window, window_ride_measurements_paint() gets called for the first time, but the ride's excitement rating is still -1 at this point. This causes the code at https://github.com/OpenRCT2/OpenRCT2/blob/develop/src/openrct2/windows/ride.c#L5246 to access RatingNames[-1].

**Steps to reproduce:**
1. Enable the -fsanitize=address compiler option. This causes ASAN to perform runtime memory and bounds checking.
2. Compile the game
3. Run the game, and enter the Roller Coaster Designer
4. Build any roller coaster, and test it with the measurements and test data tab of the ride window open.
5. When the ride has gone through a complete loop, the address sanitizer will close the game.

**Dump file**

**Screenshots / Video:**

**Save game:**

**Comment by duncanspumpkin** [comment]
Yeah it access invalid memory but it doesn't perform anything on that memory so it is at least not going to cause crashes.

**Comment by IntelOrca** [comment]
It can still crash if its reading invalid memory.

**Comment by janisozaur** [comment]
Problems start when it's trying to right-shift a signed negative number, which is undefined behaviour.
